{{cookiecutter.project_slug}}/pyproject.toml

```diff
@@ -62,19 +69,117 @@ allow-direct-references = true
 
 [tool.hatch.version]
 source = "vcs"
-raw-options = { version_scheme = "no-guess-dev" }
 
-[tool.hatch.build.hooks.vcs]
-version-file = "src/{{ cookiecutter.pkg_name }}/_version.py"
+[tool.hatch.version.raw-options]
+version_scheme = "no-guess-dev"
+local_scheme = "dirty-tag"
 
 [tool.hatch.build]
 packages = ["src/{{ cookiecutter.pkg_name }}"]
 
+[tool.hatch.build.hooks.vcs]
+version-file = "src/{{ cookiecutter.pkg_name }}/_version.py"
+
 [tool.hatch.build.targets.sdist]
+artifacts = ["_version.py"]
 exclude = [
     "/.github",
 ]
 
+[tool.bumpversion]
```

Comment:
@dornech, how does bumpversion play together with setuptools_scm? The latter takes the info from git, bumpversion uses a different concept, right?

Reply:
I am not sure if this is relevant. I do not really use setuptools_scm any more and also removed it as dependency for the "footgun" as dicsussed earlier. Setuptools_scm does not work with hatch / pyproject.toml because it does not get the raw-options.
I use bump-my-version (with calling generate-changelog as hook) or semantic-release to bump the version and generating a changelog automatically (what setuptools_scm does not do as far as I understood). What I wanted to achieve ist that the changelog is generated locally and together with creating a version before pushing to github and that works with these two toolchain amendments. The advantage here is also that with conventional commits (enforced by gitlint) this really works automatically.
What one could do: optional cookiecutter to use one or the other toolchain amendment - i. e. bump-my-version/generate-changelog or semantic-release. I have no real picture which one is better, both are pretty long "on the market" and properly maintained. handling is also pretty straightforward.